src/services/storage/index.ts

```diff
@@ -37,20 +45,50 @@ export class Storage implements StorageDriver {
 			return this.storage._driver
 		}
 		public async get(id: string): Promise<StoredUrl> {
-			return this.driver.url.get(id)
+			try {
+				logger.debug(`Running Storage.url.get with ${id}`)
+				return this.driver.url.get(id)
+			} catch (err) {
+				logger.error(`Storage.url.get failed: ${err}`)
+				throw new GeneralError('Could not get (Storage.url.get)')
+			}
 		}
 		public async delete(id: string): Promise<void> {
-			return this.driver.url.delete(id)
+			try {
+				logger.debug(`Running Storage.url.delete with ${id}`)
+				return this.driver.url.delete(id)
+			} catch (err) {
+				logger.error(`Storage.url.delete failed: ${err}`)
+				throw new GeneralError('Could not delete (Storage.url.delete)')
+			}
 		}
 		public async deleteOverdue(timespanMs: number): Promise<number> {
-			return this.driver.url.deleteOverdue(timespanMs)
+			try {
+				logger.debug(`Running deleteOverdue with ${timespanMs}`)
+				return await this.driver.url.deleteOverdue(timespanMs)
+			} catch (err) {
+				logger.error(`Storage.url.deleteOverdue failed: ${err}`)
+				throw GeneralError('Could not delete overdue')
+			}
 		}
 		public async edit(id: string, url: string): Promise<StoredUrl> {
-			return this.driver.url.edit(id, url)
+			try {
+				logger.debug(`Running Storage.url.edit with ${id} and ${url}`)
+				return this.driver.url.edit(id, url)
+			} catch (err) {
+				logger.error(`Storage.url.edit failed: ${err}`)
+				throw new GeneralError('Could not edit (Storage.url.edit)')
+			}
 		}
 
 		public async save(url: string): Promise<StoredUrl> {
-			return this.driver.url.save(url)
+			try {
+				logger.debug(`Start Storage.url.save with ${url}`)
+				return this.driver.url.save(url)
+			} catch (err) {
+				logger.error(`Storage.url.save failed: ${err}`)
+				throw new GeneralError('Could not save (Storage.url.save)')
```

Comment:
Remember that these errors are returned to the client, so don't add the function name here.
```suggestion
				throw new GeneralError('Could not save url')
```